src/context/WalletProvider/Ledger/components/Connect.tsx

```diff
@@ -0,0 +1,83 @@
+import { WebUSBLedgerAdapter } from '@shapeshiftoss/hdwallet-ledger-webusb'
+import React, { useState } from 'react'
+import type { RouteComponentProps } from 'react-router-dom'
+import type { ActionTypes } from 'context/WalletProvider/actions'
+import { WalletActions } from 'context/WalletProvider/actions'
+import { KeyManager } from 'context/WalletProvider/KeyManager'
+import { setLocalWalletTypeAndDeviceId } from 'context/WalletProvider/local-wallet'
+import { useWallet } from 'hooks/useWallet/useWallet'
+
+import { ConnectModal } from '../../components/ConnectModal'
+import { LedgerConfig } from '../config'
+
+export interface LedgerSetupProps
+  extends RouteComponentProps<
+    {},
+    any // history
+  > {
+  dispatch: React.Dispatch<ActionTypes>
+}
+
+export const LedgerConnect = ({ history }: LedgerSetupProps) => {
+  const { dispatch: walletDispatch, state } = useWallet()
+  const [loading, setLoading] = useState(false)
+  const [error, setError] = useState<string | null>(null)
+
+  // eslint-disable-next-line no-sequences
+  const setErrorLoading = (e: string | null) => (setError(e), setLoading(false))
+
+  const pairDevice = async () => {
+    setError(null)
+    setLoading(true)
+    if (state.adapters) {
+      const currentAdapters = state.adapters
+      // eslint is drunk, this isn't a hook
+      // eslint-disable-next-line react-hooks/rules-of-hooks
+      const ledgerAdapter = WebUSBLedgerAdapter.useKeyring(state.keyring)
+      // This is conventionally done in WalletProvider effect, but won't work here, as `requestDevice()` needs to be called from a user interaction
+      // So we do it in this pairDevice() method instead and set the adapters the same as we would do in WalletProvider
+      try {
+        const wallet = await ledgerAdapter.pairDevice()
+        try {
+          currentAdapters.set(KeyManager.Ledger, [ledgerAdapter])
+          walletDispatch({ type: WalletActions.SET_ADAPTERS, payload: currentAdapters })
+        } catch (e) {
+          console.error(e)
+        }
+
+        if (!wallet) {
+          setErrorLoading('walletProvider.errors.walletNotFound')
+          throw new Error('Call to hdwallet-ledger::pairDevice returned null or undefined')
+        }
+
+        const { name, icon } = LedgerConfig
+        // TODO(gomes): this is most likely wrong, all Ledger devices get the same device ID
```

Comment:
Will bring `ledger-<uuid>` as a follow-up

Reply:
Or perhaps not - given they all use `0001` as a serial number, the uuid trick won't cut it either as it would generate a new deviceId every connection. We'll have to go with the fact we can't identify a given Ledger and build around that UX-wise.
`ledger-0001` would be it